File: MiniDOMBase/DMD/Databases/Old/Struttura/CVS/CCvsDBConnection.vb
Imports minidom
Imports minidom.Sistema
Imports System.Xml.Serialization

Public partial class Databases

    Public Class CCvsDBConnection
        Inherits CXlsDBConnection

       
        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal fileName As String, Optional ByVal useHeaders As Boolean = True)
            MyBase.New(fileName, "", useHeaders)
        End Sub


        Public Overrides Function GetSqlDataType(field As CDBEntityField) As String
            If field.DataType Is GetType(String) Then
                Return IIf(field.MaxLength > 0, "TEXT(255)", "MEMO")
            Else
                Return MyBase.GetSqlDataType(field)
            End If
        End Function

        Public Overrides Function GetFriendlyName(name As String) As String
            name = MyBase.GetFriendlyName(name)
            If Right(name, 1) = "$" Or Right(name, 1) = "_" Then
                Return Left(name, Len(name) - 1)
            Else
                Return name
            End If
        End Function

        Public Overrides Function GetInternalTableName(table As CDBEntity) As String
            Return Replace(table.Name, "#c", ".csv")
        End Function

        Public Overrides Function GetConnectionString() As String
            Const excelObject As String = "Provider=Microsoft.{0}.OLEDB.{1};Data Source={2};Extended Properties=""{3};HDR=YES;{4}"""
            Return String.Format(excelObject, "Jet", "4.0", Me.Path, "text", "FMT=Delimited")
        End Function

        Protected Overrides Function GetTablesArray() As CDBTable()
            Dim ret As CDBTable() = MyBase.GetTablesArray()
            For i As Integer = 0 To UBound(ret)
                Dim table As CDBTable = ret(i)
                table.IsHidden = Right(table.Name, 1) = "_"
                table.Name = Left(table.Name, Len(table.Name) - 1)
            Next
            Return ret
        End Function

        

    End Class


End Class

File: MiniDOMBase/DMD/Databases/Old/Struttura/CVS/CCvsDBConnection.cs


namespace minidom
{
    public partial class Databases
    {
        public class CCvsDBConnection 
            : CXlsDBConnection
        {
            public CCvsDBConnection() : base()
            {
            }

            public CCvsDBConnection(string fileName, bool useHeaders = true) : base(fileName, "", useHeaders)
            {
            }

            public override string GetSqlDataType(CDBEntityField field)
            {
                if (ReferenceEquals(field.DataType, typeof(string)))
                {
                    return Sistema.IIF(field.MaxLength > 0, "TEXT(255)", "MEMO");
                }
                else
                {
                    return base.GetSqlDataType(field);
                }
            }

            public override string GetFriendlyName(string name)
            {
                name = base.GetFriendlyName(name);
                if (DMD.Strings.Right(name, 1) == "$" | DMD.Strings.Right(name, 1) == "_")
                {
                    return DMD.Strings.Left(name, DMD.Strings.Len(name) - 1);
                }
                else
                {
                    return name;
                }
            }

            public override string GetInternalTableName(CDBEntity table)
            {
                return DMD.Strings.Replace(table.Name, "#c", ".csv");
            }

            public override string GetConnectionString()
            {
                const string excelObject = "Provider=Microsoft.{0}.OLEDB.{1};Data Source={2};Extended Properties=\"{3};HDR=YES;{4}\"";
                return string.Format(excelObject, "Jet", "4.0", Path, "text", "FMT=Delimited");
            }

            protected override CDBTable[] GetTablesArray()
            {
                var ret = base.GetTablesArray();
                for (int i = 0, loopTo = DMD.Arrays.UBound(ret); i <= loopTo; i++)
                {
                    var table = ret[i];
                    table.IsHidden = DMD.Strings.Right(table.Name, 1) == "_";
                    table.Name = DMD.Strings.Left(table.Name, DMD.Strings.Len(table.Name) - 1);
                }

                return ret;
            }
        }
    }
}